### ISSUE TITLE
[Feature Request] Add Last filter

### ISSUE THREAD

**Issue by udithprabhu** [bitfield/script#18]
Add Last filter which is analogous to tail command

**Comment by bitfield** [comment]
Thank you! Well, you know the drill: please write an example program which uses the feature, and we'll get to work!

**Comment by udithprabhu** [comment]
Last can be used to get the last 10 errors as below

`script.Stdin().Match("error").Last(10).Stdout()`

**Comment by bitfield** [comment]
That's perfectly fine, but not quite what I had in mind. _Why_ would you want this? If you're working on the command line, reading log files, you presumably have `tail` or `less` available. But why would a _program_ want the last X lines of something?

**Comment by udithprabhu** [comment]
One of the usecases that I had was to perform an analysis on recent logs rather than all the logs present in a log file. As you said we can use tail and pipe the result to the go program , but I feel that having this behaviour in the package itself will offer more flexibility.

One more usecase can be that Last paired with Freq can be used to get the least frequent lines.

**Comment by bitfield** [comment]
Analysing 'recent logs' probably needs something a bit more sophisticated than just the last X lines, doesn't it? We wouldn't want to read the same log lines twice, for example.

> One more usecase can be that Last paired with Freq can be used to get the least frequent lines.

Of what, and why? I'm not being purposefully obtuse; I'm trying to dig into what the _use case_ is, which isn't the same as 'an example of using it'. A use case is some real task you need to do.

**Comment by udithprabhu** [comment]
One of the use cases that I encountered recently in my company requires such analysis. The use case was to log the least used control flow path in a critical function, I had thought of using Freq to find it , looking at its implementation , I found that it is sorted in descending order, so with Last the entire implementation can be contained in one go program , otherwise I would have to use tail the output from the go program.

This is what urged me to create the Issue.

One more use case is that finding the Nth line of input such as 

`
p.First(n).Last(1).String()
`

This is present as tail and head in a shell script that I wanted to port to go using scripts

But as the current implementation did not have Last ( at least not a straightforward one ) , I could not port it successfully.

I cannot go more indepth of its needs as the use cases that I have presented are too specific to my organisation, but I assume that such use cases are present for majority of the organisations.
